src/store.ts

```diff
@@ -36,13 +36,29 @@ export class StateStore<T extends Record<string, unknown>> {
     };
   };
 
-  public subscribeWithSelector = <O extends readonly unknown[]>(selector: (nextValue: T) => O, handler: Handler<O>) => {
+  public subscribeWithSelector = <O extends Readonly<Record<string, unknown>> & { length?: never }>(
```

Comment:
```suggestion
  public subscribeWithSelector = <O extends Readonly<Record<string, unknown>>>(
```
Let's not keep this here; it was fine as a temporary measure (to easily find all the selectors and change outputs to objects).

Reply:
I thought about removing it, but I'd actually rather keep it so that TS complains on the integrator's side as well (it won't otherwise). They'll get the warning, sure - but it won't "force" them to change; if they use TS this rule would.

Reply:
Nevermind, the SDKs will anyway be complaining about this. Will remove it, thanks.